src/lib.rs

```diff
@@ -1 +1,139 @@
+use windows::Win32::Foundation::{CloseHandle, GetLastError, PWSTR};
+use windows::Win32::Security::SECURITY_ATTRIBUTES;
+use windows::Win32::System::Threading::{
+    TerminateProcess, WaitForSingleObject, PROCESS_CREATION_FLAGS, PROCESS_INFORMATION,
+    STARTUPINFOW,
+};
+use windows::Win32::System::WindowsProgramming::INFINITE;
+
+#[derive(Debug)]
+pub struct ChildProcess {
+    command: String,
+    process_information: PROCESS_INFORMATION,
+}
+
+impl ChildProcess {
+    pub fn new(
+        command: &str,
+        inherit_handles: bool,
+        current_directory: Option<impl AsRef<Path>>,
+    ) -> Result<Self, ChildProcessError> {
+        unsafe {
+            let mut si = STARTUPINFOW::default();
+            let mut pi = PROCESS_INFORMATION::default();
+
+            si.cb = size_of::<STARTUPINFOW>() as u32;
+
+            let process_creation_flags = PROCESS_CREATION_FLAGS(0);
+
+            let res = if let Some(directory) = current_directory {
+                let directory = directory.as_ref().as_os_str();
+                windows::Win32::System::Threading::CreateProcessW(
+                    PWSTR::default(),
+                    command,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    directory,
+                    &si,
+                    &mut pi as *mut PROCESS_INFORMATION,
+                )
+            } else {
+                windows::Win32::System::Threading::CreateProcessW(
+                    PWSTR::default(),
+                    command,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    PWSTR::default(),
+                    &si,
+                    &mut pi as *mut PROCESS_INFORMATION,
+                )
+            };
+
+            if res.as_bool() {
+                Ok(Self {
+                    command: command.to_string(),
+                    process_information: pi,
+                })
+            } else {
+                Err(ChildProcessError::CreationFailed(format!(
+                    "{:?}",
+                    GetLastError()
+                )))
+            }
+        }
+    }
+
+    pub fn wait(&self) -> ExitStatus {
+        unsafe {
+            let exit_code = WaitForSingleObject(self.process_information.hProcess, INFINITE);
+            CloseHandle(self.process_information.hProcess);
+            CloseHandle(self.process_information.hThread);
+
+            ExitStatus(exit_code)
+        }
+    }
+
+    pub fn try_wait(&self, duration: u32) -> Result<ExitStatus, ExitStatusError> {
```

Comment:
1.  This return type is wrong. `try_wait` means you will or will not receive an exit status. 

    If the process has exited before you call try_wait, then you should receive an ExitStatus.

    If the process is still running, you can't get an exit status since it has not exited.

    Check std's API documentation: https://doc.rust-lang.org/std/process/struct.Child.html#method.try_wait

2.  `duration` should be a std::time::Duration because you are making a wrapper library that makes this easy to use with Rust. It's exactly like using String instead of Path.